### ISSUE TITLE
How can I test a site that has a password on it?

### ISSUE THREAD

**Issue by ashrobbins** [garris/BackstopJS#1124]
Hi, I want to use Backstop on my dev and test sites, but the sites are password protected before you even get to them. Is this possible?

As far as I can tell the other issues in this repo are talking about site login, whereas my issue is more like server access I believe.

Can I do this with a cookie or by using `onBefore.js`?

**Comment by garris** [comment]
You can do this with the onReady script.  Search google for puppeteer authentication api.

**Comment by Romancho** [comment]
For Base Auth you can look athttps://github.com/Romancho/backstopjs-tools/backstop_data/engine_scripts/puppet/onBefore.js

**Comment by soulstyle** [comment]
I've been struggling with this for the last 3 days. I would appreciate some input on this as I don't know what more to test...

I'm using the `await page.authenticate({username: config.stageHttpAuthUser, password: config.stageHttpAuthPass});` as in @Romancho example.

If I console log the `page` variable I can see `_credentials: { username: 'xxxxxxxx', password: 'xxxxxxx' },` in the `_networkManager` object, which is correct, but for some reason it still doesn't authenticate.

If I run the script with `"debugWindow": true` I can see the un-authenticated page.

I'm running the default scenario that just opens the homepage of a site. The reference URL is without HTTP Auth and the test URL is with auth. The _reference_ screenshots are working fine but not the _test_ screenshots where it just shows a white 401 Authorization Required page.

**What I've tried so far:**
- set page.authenticate in onBefore.js
- set page.setExtraHTTPHeaders in onBefore.js
- add auth header manually with interceptedRequest

**Files:**
backstop-config: https://www.codepile.net/pile/LkMz3o9W
onBefore.js: https://www.codepile.net/pile/1kr1jqxM (including other things I've tried commented out)
Script output including console.log: https://www.codepile.net/pile/gz95dOze

Thanks a lot!
